### DISCUSSION TITLE
Support TERM_FEATURES specification

### DISCUSSION THREAD

**Discussion by gpanders** [ghostty-org/ghostty#3082] [Feature Requests, Ideas] [answered]
iTerm has proposed a `TERM_FEATURES` environment variable (and potentially control sequence) for reporting terminal capabilities: https://iterm2.com/feature-reporting/.

Should Ghostty implement this as well? Wanted to open an issue for discussion.

**Pros**:

This is a backward compatible method of declaring terminal capabilities that does not suffer the drawbacks of terminfo/`$TERM`. Using an environment variable would still require users to forward the env var over their SSH connection, but would not require copying terminfo entries. If/when an escape sequence is supported, then not even that would be necessary (though of course would still require support from applications).

**Cons**

It's not clear how much support this proposal has (I was personally not even aware of it until today). I am not even sure if iTerm2 itself implements it yet.

cc @rockorager

**Comment by mitchellh** [comment]
I don't see any real downside to supporting this but to ask the obvious: does anyone use this? do any other terminals implement this? I'm a huge fan of XTGETTCAP but I can see why this would be a lower barrier to entry for many programs.

**Comment by mitchellh** [comment]
Converted to a feature request discussion to gather some more feedback on this. As I noted, I'm not against it, just not sure if its worth it.

**Comment by mitchellh** [comment, marked as answer]
Going to mark this as "no" right now. If someone PRs it then I'm fine with it but I don't see any demand or usage of this and I feel like other specs solve this current.
